### PR TITLE
Add fields connected to rules and alerts

### PR DESCRIPTION
**Pull request by tomsonpl**
Closes: https://github.com/elastic/security-team/issues/5929

Not sure if we need rule_name, but from what I remember, it was agreed that we wanted it.

### CODE REVIEW

**Comment by tomsonpl** [comment]
@pzl any idea if `rule_name` should be a `keyword` or just `text` ?

**Comment by pzl** [comment]
> any idea if `rule_name` should be a `keyword` or just `text` ?

are we ever going to filter/search via name? I would imagine in filtering scenarios we probably know the rule_id and will be exact matching on that, right? 

If for some reason we will filter on the name, do we know the full string to match (e.g. find rule by name "Unusual Network Activity")? then keyword. If we expect to filter on partial word matches (e.g. match the previous rule name based on the search string "network") then text could be ok.

If we don't need to filter by name, then I would set it to keyword, and `index: false`. Or just don't map it at all

**Comment by tomsonpl** [comment]
@pzl @dasansol92 could you take a look please? Thank you!

**Review by pzl — APPROVED** [review]
looks okay, I think things are added in the right place. Added @ashokaditya to review the intent of whats being added

File: custom_schemas/custom_endpoint_actions.yml [hunk 1]
```diff
@@ -35,6 +35,13 @@
       short: data
       description: >
         The action request information
+
+    - name: data.alert_ids
+      type: keyword
+      level: custom
+      short: alert ids
+      description: >
+        List of alert ids that triggered the action
```
Comment (ashokaditya) [inline custom_schemas/custom_endpoint_actions.yml:44]: You'd also need to edit the `custom_subsets/elastic_endpoint/actions/actions.yaml` file. If you want the fields to also exist on the response doc then you need to edit the `custom_subsets/elastic_endpoint/action_responses/action_responses.yaml`.

**Review by ashokaditya — CHANGES_REQUESTED** [review]
I added some suggestions to update the mappings. Let me know if you need help or have questions.

**Review by ashokaditya — COMMENTED** [review]
Took another pass at reviewing. I have a question and suggestions.

File: custom_schemas/custom_endpoint_actions.yml [hunk 2]
```diff
@@ -35,6 +35,13 @@
       short: data
       description: >
         The action request information
+
+    - name: data.alert_id
+      type: keyword
+      level: custom
+      short: alert id
+      description: >
+        List of alert ids that triggered the action
```
Comment (ashokaditya) [inline custom_schemas/custom_endpoint_actions.yml:44]: Thanks for the changes. The fact that this is a `keyword` we can store an array of strings here s, so you can have multiple alert ids in the doc. Just that we should conform to ECS naming best practices.

**Review by ashokaditya — APPROVED** [review]

**Review by ashokaditya — APPROVED** [review]

**Comment by tomsonpl** [comment]
@ashokaditya Thank you for your feedback and bearing with me fixing it so many times :P

**Pull request merged**